app/logic/Mail/ActiveSync/ActiveSyncAccount.ts

```diff
@@ -441,15 +441,15 @@ export class ActiveSyncAccount extends MailAccount {
             break;
           case FolderType.Calendar:
           case FolderType.UserCalendar:
-            // TODO compare calendar ID
-            let calendar = appGlobal.calendars.find((calendar: ActiveSyncCalendar) => calendar.mainAccount == this) as ActiveSyncCalendar | null;
+            let calendar = appGlobal.calendars.find(calendar => calendar.mainAccount == this && calendar.url == url.toString()) as ActiveSyncCalendar | null;
```

Comment:
Aha, I had been wondering about this!

Reply:
Yes, putting the serverID into the URL was mighty subtle!

We should put it into a separate field. It wasn't possible before, but now with the JSON properties, we can store arbitrary data for each account.

Reply:
Thanks for having noticed this. This is the level of review that I needed.